supervision/metrics/intersection_over_union.py

```diff
@@ -0,0 +1,136 @@
+        data_2: Union[npt.NDArray, Detections],
+    ) -> Metric:
+        """
+        Add data to the metric, without computing the result.
+
+        The arguments can be:
+        * Boxes of shape (N, 4), float32,
+        * Masks of shape (N, H, W), bool
+        * Oriented bounding boxes of shape (N, 8), float32.
+        * Detections object.
+
+        Args:
+            data_1 (Union[npt.NDArray, Detection]): The first set of data.
+            data_2 (Union[npt.NDArray, Detection]): The second set of data.
+
+        Returns:
+            Metric: The metric object itself. You can get the metric result
+            by calling the `compute` method.
+        """
+        self._store.update(data_1, data_2)
+        return self
+
+    def compute(self) -> Dict[int, npt.NDArray[np.float32]]:
+        """
+        Compute the Intersection over Union metric (IoU)
+        Uses the data set with the `update` method.
+
+        Returns:
+            Dict[int, npt.NDArray[np.float32]]: A dictionary with class IDs as keys.
+            If no class ID is provided, the key is the value CLASS_ID_NONE.
+        """
+        # TODO: cache computed result.
+        ious = {}
+        for class_id, array_1, array_2 in self._store:
+            if self._metric_target == MetricTarget.BOXES:
+                if array_1 is None or array_2 is None:
+                    ious[class_id] = np.empty((0, 4), dtype=np.float32)
+                    continue
+                iou = self._compute_box_iou(array_1, array_2)
+
+            else:
+                raise NotImplementedError(
+                    "Intersection over union is not implemented"
+                    " for {self._metric_target}."
+                )
+            ious[class_id] = iou
+        return ious
+
+    # TODO: This would return dict[int, pd.DataFrame]. Do we want that?
+    #       It'd be cleaner if it returned a single DataFrame, but the sizes
+    #       differ if class_agnostic=False.
+
+    # def to_pandas(self) -> 'pd.DataFrame':
+    #     """
+    #     Return a pandas DataFrame representation of the metric.
+    #     """
+    #     self._ensure_pandas_installed()
+    #     import pandas as pd
+
+    #     # TODO: use cached results
+    #     ious = self.compute()
+    #     print(len(ious))
+
+    #     class_names = []
+    #     arrays = []
+
+    #     for class_id, array in ious.items():
+    #         print(array.shape)
+    #         class_names.append(np.full(array.shape[0], class_id))
+    #         arrays.append(array)
+    #     stacked_class_ids = np.concatenate(class_names)
+    #     stacked_ious = np.vstack(arrays)
+    #     combined = np.column_stack((stacked_class_ids, stacked_ious))
+
+    #     column_names = \
+    # ['class_id'] + [f'col_{i+1}' for i in range(stacked_ious.shape[1])]
+    #     result = pd.DataFrame(combined, columns=column_names)
+
+    #     return result
+
+    @staticmethod
+    def _compute_box_iou(
```

Comment:
Why do we need to duplicate the logic from `box_iou_batch`?

Reply:
Removed for now.

My initial idea was to replace `box_iou_batch` with the metric. That's still on the table, but let's talk about it when the metric is complete.